tests/triggers_tests.cpp

```diff
@@ -510,6 +510,83 @@ TEST_CASE("effect scope absorbsion", "[effect_tests]") {
 	}
 }
 
+TEST_CASE("effect op fusion", "[effect_tests]") {
+	{
+		std::vector<uint16_t> t;
+		t.push_back(uint16_t(effect::generic_scope));
+		t.push_back(uint16_t(5));
+		t.push_back(uint16_t(effect::clr_global_flag));
+		t.push_back(uint16_t(4));
+		t.push_back(uint16_t(effect::clr_global_flag));
+		t.push_back(uint16_t(7));
+		const auto new_size = parsers::simplify_effect(t.data());
+		REQUIRE(5 == new_size);
+		REQUIRE(t[2] == uint16_t(effect::fop_clr_global_flag_2));
+		REQUIRE(t[3] == uint16_t(4));
+		REQUIRE(t[4] == uint16_t(7));
+	}
+	{
+		std::vector<uint16_t> t;
+		t.push_back(uint16_t(effect::generic_scope));
+		t.push_back(uint16_t(7));
+		t.push_back(uint16_t(effect::clr_global_flag));
+		t.push_back(uint16_t(4));
+		t.push_back(uint16_t(effect::clr_global_flag));
+		t.push_back(uint16_t(7));
+		t.push_back(uint16_t(effect::clr_global_flag));
```

Comment:
add a non-clr global flag effect before and after this run of three to make sure that the simplification isn't interfering with other effects in the same scope